heartbeat/sender/routineHandler.go

```diff
@@ -49,6 +52,7 @@ func (handler *routineHandler) processLoop(ctx context.Context) {
 			handler.heartbeatSender.Execute()
 		case <-handler.hardforkSender.ShouldTriggerHardfork():
 			handler.hardforkSender.Execute()
+			time.Sleep(handler.delayAfterHardforkMessageBroadcast)
```

Comment:
Do not use time.Sleep here but rather a select that takes into account the context to properly finish the go routine on this case.

Reply:
done, added an unit test for this.